src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.cloudsdk;
+
+import com.google.cloud.tools.appengine.AppEngineDescriptor;
+import com.google.cloud.tools.appengine.api.AppEngineException;
+import com.google.cloud.tools.appengine.api.devserver.AppEngineDevServer;
+import com.google.cloud.tools.appengine.api.devserver.RunConfiguration;
+import com.google.cloud.tools.appengine.api.devserver.StopConfiguration;
+import com.google.cloud.tools.appengine.cloudsdk.internal.args.DevAppServerArgs;
+import com.google.cloud.tools.appengine.cloudsdk.internal.process.ProcessRunnerException;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Classic Java SDK based implementation of {@link AppEngineDevServer}.
+ */
+public class CloudSdkAppEngineDevServer1 implements AppEngineDevServer {
+
+  private final CloudSdk sdk;
+
+  private static final String DEFAULT_HOST = "localhost";
+  private static final int DEFAULT_PORT = 8080;
+
+  public CloudSdkAppEngineDevServer1(CloudSdk sdk) {
+    this.sdk = Preconditions.checkNotNull(sdk);
+  }
+
+  /**
+   * Starts the local development server, synchronously or asynchronously.
+   *
+   * @throws AppEngineException I/O error in the Java dev server.
+   */
+  @Override
+  public void run(RunConfiguration config) throws AppEngineException {
+    Preconditions.checkNotNull(config);
+    Preconditions.checkNotNull(config.getServiceDirectories());
+    Preconditions.checkArgument(config.getServiceDirectories().size() > 0);
+    AppEngineDescriptor appengineWeb;
+    try {
+      // TODO(ludo: Make sure we support the case when more than 1 service is given...
+      FileInputStream is = new FileInputStream(
+              new File(config.getServiceDirectories().get(0), "WEB-INF/appengine-web.xml"));
```

Comment:
I don't think `AppEngineDescriptor.parse()` closes `InputStream`. Suggesting to use a try-with-resources block.